### PR TITLE
Add treasure hunt resource and fixes to resource generation

### PR DESCRIPTION
**Pull request by JackMorganNZ**
Related: #18 #81 #82

### CODE REVIEW

File: csunplugged/resources/views/sorting_network.py [hunk 1]
```diff
@@ -49,3 +37,37 @@ def resource_image(get_request, resource):
...
     return image
+
+
+def subtitle(get_request, resource):
+    """Returns the subtitle string of the resource.
+
+    Used after the resource name in the filename, and
+    also on the resource image.
+    """
+    TEMPLATE = '{} to {}'
```
Comment (hayleyavw) [inline csunplugged/resources/views/sorting_network.py]: you're a little inconsistent with naming template strings.
Please make this more descriptive than just "template"

File: csunplugged/static/scss/website.scss [hunk 2]
```diff
@@ -20,3 +20,6 @@ h1, h2, h3 {
     color: #888;
   }
 }
+input[type="text"].long-text-field {
+  width: 20em;
+}
```
Comment (hayleyavw) [inline csunplugged/static/scss/website.scss]: missing newline (from my commit), you might as well fix before merging into develop

File: csunplugged/templates/resources/treasure-hunt.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{% extends "resources/base-resource.html" %}
+
+<!-- Name of resource -->
+{% block name %}{{ resource.name }}{% endblock name %}
```
Comment (hayleyavw) [inline csunplugged/templates/resources/treasure-hunt.html]: happy with opening and closing a block on the same line? Or should there be a new line here for consistency? (see lines 7 and 8)

Comment (JackMorganNZ) [inline csunplugged/templates/resources/treasure-hunt.html]: Was to match usage of parent, see base template. We could always write as block as whitespace is irrelevant?

File: csunplugged/resources/views/sorting_network.py (hunk 1 above)
Comment (JackMorganNZ) [inline csunplugged/resources/views/sorting_network.py]: Suggestions?

**Comment by JackMorganNZ** [comment]
I'm hopeless at remembering to go back and update docstrings after I've written the function. Hopefully I'll learn, but would it be wise to install a style checker to check each function/method has a Google style docstring? That would reduce the mental load of having us to remember?

File: csunplugged/resources/views/sorting_network.py (hunk 1 above)
Comment (JackMorganNZ) [inline csunplugged/resources/views/sorting_network.py]: `SUBTITLE_TEMPLATE`?

Comment (hayleyavw) [inline csunplugged/resources/views/sorting_network.py]: much better

File: csunplugged/templates/resources/treasure-hunt.html (hunk 3 above)
Comment (hayleyavw) [inline csunplugged/templates/resources/treasure-hunt.html]: Maybe we should talk about this when you go through all the templates (re your comment on the scratch PR)

**Comment by hayleyavw** [comment]
The problem is probably them becoming out of date quickly rather than not being there at all (in saying that I just found I missed one entirely in kordac :joy:), I think we just need to be more diligent and less lenient on  PR reviews

**Pull request merged**